lib/features/pokemon_details/pokemon_details_page.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+import 'package:pokedex/features/pokemon_details/pokemon_about/pokemon_about_page.dart';
+import 'package:pokedex/features/pokemon_details/pokemon_moves/pokemon_moves_page.dart';
+import 'package:pokedex/features/pokemon_details/pokemon_stats/pokemon_stats_page.dart';
+import 'package:pokedex/features/pokemon_overview/pokemon_overview_connector.dart';
+import 'package:pokedex/utils/strings.dart' as str;
+
+class PokemonDetailsPage extends StatelessWidget {
+  const PokemonDetailsPage({
+    Key? key,
+  }) : super(key: key);
+
+  static const String route = '/details';
+  static const String routeName = 'details';
+
+  @override
+  Widget build(BuildContext context) {
+    final appBarViews = <Widget>[
+      const PokemonAboutPage(),
+      const PokemonStatsPage(),
+      const PokemonMovesPage(),
+    ];
+    final appBarItems = <Tab>[
+      const Tab(child: Text(str.aboutTitle)),
+      const Tab(child: Text(str.statsTitle)),
+      const Tab(child: Text(str.movesTitle)),
+    ];
+    return DefaultTabController(
+      length: appBarItems.length,
+      child: Scaffold(
+        appBar: AppBar(
+          leading: IconButton(
```

Comment:
this isn't needed anymore - let's maximize the native behavior here. Kindly supply the proper sub_route then it will automatically provide backStack button for you 😜